x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_upgrade_notifications.md

```diff
@@ -0,0 +1,331 @@
+And user should NOT see a number of rules available to upgrade
+When user waits until the the package installation is completed
+Then user should see the callout to upgrade prebuilt rules
+And user should see the Rule Updates tab
+And user should see the number of rules available to upgrade (<Z>)
+
+Examples:
+  | X    | Z    |
+  | 1250 | 1    |
+  | 1250 | 100  |
+  | 1250 | 1250 |
+```
+
+#### **Scenario: User is notified on the Rule Management page when there are some prebuilt rules to upgrade but there are no more prebuilt rules to install**
+
+**Automation**: 1 e2e test with mock rules + 1 integration test with mock rules for the /status endpoint.
+
+```Gherkin
+Given <X> prebuilt rules are installed in Kibana
+And there are no more prebuilt rules available to install
+And for <Z> of the installed rules there are new versions available
+When user opens the Rule Management page
+Then user should see the callout to upgrade prebuilt rules
+And user should see the Rule Updates tab
+And user should see the number of rules available to upgrade (<Z>)
+
+Examples:
+  | X    | Z    |
+  | 1250 | 1    |
+  | 1250 | 100  |
+  | 1250 | 1250 |
+```
+
+#### **Scenario: User is notified on the Rule Management page when there are some prebuilt rules to upgrade and some more prebuilt rules to install**
+
+**Automation**: 1 e2e test with mock rules + 1 integration test with mock rules for the /status endpoint.
+
+```Gherkin
+Given <X> prebuilt rules are installed in Kibana
+And there are <Y> more prebuilt rules available to install
+And for <Z> of the installed rules there are new versions available
+When user opens the Rule Management page
+Then user should see the callout to upgrade prebuilt rules
+And user should see the Rule Updates tab
+And user should see the number of rules available to upgrade (<Z>)
+
+Examples:
+  | X    | Y    | Z    |
+  | 1    | 1249 | 1    |
+  | 1249 | 1    | 100  |
+  | 1249 | 1    | 1249 |
+```
+
+#### **Scenario: User can open the Rule Upgrade table on the Rule Management page**
+
+**Automation**: 1 e2e test with mock rules.
+
+```Gherkin
+Given some prebuilt rules are installed in Kibana
+And for some of the installed rules there are new versions available
+When user opens the Rule Management page
+Then user should see the Rule Updates tab
+When user clicks on the Rule Updates tab
+Then the Rule Upgrade table should be displayed
+```
+
+#### **Scenario: User can dismiss the prebuilt rule upgrade callout on the Rule Management page**
+
+**Automation**: 1 e2e test with mock rules + unit tests for the callout component.
+
+```Gherkin
+Given some prebuilt rules are installed in Kibana
+And for some of the installed rules there are new versions available
+When user opens the Rule Management page
+Then user should see the callout to upgrade prebuilt rules
+When user clicks on the Dismiss button
+Then the callout should disappear
+When user refreshes the page
+Then user should NOT see the callout
```

Comment:
Currently, when a user refreshes the page, this callout reappears, are we wanting to change this logic?

Reply:
Yes! Thanks for the catch.